<h4 align="center" style="height:1123px;font-family: 'Times New Roman'"><b>

    LAPORAN PRAKTIKUM

    ALGORITMA & PEMOGRAMAN

    “OPERATOR PADA PYTHON”

    Disusun Oleh :

    Muhammad Dzaky Yazid

    2611531011

    Dosen Pengampu:

    Wahyudi DR.Ir.. S.T.M.T

    Asisten Praktikum:

    Muhammad Galid Avero

<br>

 <img src="image/logo_unand.png" width="200">

<br>

    FAKULTAS TEKNOLOGI INFORMASI

    DEPARTEMEN INFORMATIKA

    UNIVERSITAS ANDALAS

    2026

</b></h4>
</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">

<h2 align="center">KATA PENGANTAR</h2>
Puji syukur kehadirat Tuhan Yang Maha Esa atas segala rahmat dan karunia-Nya sehingga laporan praktikum yang berjudul “Operasi pada Python” ini dapat diselesaikan dengan baik. Laporan ini disusun sebagai salah satu bentuk pemenuhan tugas praktikum Algoritma dan Pemrograman serta sebagai sarana untuk menambah pemahaman mengenai penggunaan operator dalam bahasa pemrograman Python.

Melalui praktikum ini, penulis memperoleh pemahaman mengenai berbagai jenis operasi dalam Python, seperti operasi aritmatika, perbandingan, logika, penugasan, dan beberapa operasi lainnya. Pemahaman tersebut diharapkan dapat menjadi dasar bagi penulis dalam mengembangkan program yang lebih sederhana, terstruktur, dan sesuai dengan kebutuhan.

Penulis menyadari bahwa dalam penyusunan laporan ini masih terdapat kekurangan, baik dari segi isi maupun penyajiannya. Oleh karena itu, kritik dan saran yang membangun sangat diharapkan untuk perbaikan di masa yang akan datang. Semoga laporan ini dapat memberikan manfaat dan menambah wawasan bagi penulis maupun pembaca dalam memahami dasar-dasar operasi pada Python.
.


<div align="right" style="margin-left:-30px;">
Padang, 2026
<br>
<br>
<br>
<br>
<br>
Muhammad Dzaky Yazid
<br>
2611531021
</div>
</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">BAB I<br>PENDAHULUAN</h1>

## 1.1 Latar Belakang
Pemrograman merupakan bagian penting dalam bidang Informatika yang digunakan untuk menyelesaikan berbagai permasalahan melalui instruksi yang dapat dijalankan oleh komputer. Dalam proses pembuatan program, operator memiliki peran penting karena digunakan untuk melakukan berbagai proses, seperti perhitungan, perbandingan, pemberian nilai, serta pengolahan kondisi.

Python menyediakan berbagai jenis operator dengan penulisan yang sederhana sehingga dapat memudahkan proses pembelajaran dasar pemrograman. Beberapa operator yang umum digunakan antara lain operator aritmatika, perbandingan, logika, penugasan, bitwise, dan operator lainnya. Setiap jenis operator memiliki fungsi dan cara penggunaan yang berbeda sesuai dengan kebutuhan dalam suatu program.

Melalui praktikum operasi pada Python, mahasiswa diharapkan dapat memahami fungsi dari berbagai jenis operator serta mengetahui cara penggunaannya dalam program. Pemahaman tersebut dapat menjadi dasar untuk mengembangkan kemampuan dalam membuat dan menyelesaikan program Python yang lebih kompleks.


## 1.2 Tujuan
1. Mengetahui berbagai jenis operasi pada Python.
2. Memahami fungsi dari setiap operator pada Python.
3. Mampu menggunakan operator dalam pembuatan program sederhana.
4. Mengetahui hasil dari setiap operasi yang dilakukan dalam Python.

## 1.3 Manfaat
1. Menambah pemahaman mengenai dasar-dasar bahasa pemrograman Python.
2. Membantu mahasiswa memahami penggunaan operator.
3. Melatih kemampuan dalam membuat program sederhana menggunakan Python.
4. Menjadi dasar untuk mempelajari pemrograman yang lebih kompleks.
</div>

---
<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">BAB II<br>PEMBAHASAN</h1>

## 2.1 Dasar Teori
Python memiliki berbagai jenis operator yang digunakan untuk melakukan proses tertentu terhadap data dan variabel. Operator merupakan simbol atau kata khusus yang digunakan untuk melakukan operasi, seperti perhitungan, perbandingan, pemberian nilai, maupun pengolahan kondisi. Penggunaan operator menjadi salah satu dasar penting dalam pemrograman karena hampir setiap program membutuhkan operasi terhadap data.

1. Operator Aritmatika
Operator aritmatika digunakan untuk melakukan operasi perhitungan matematika pada Python. Operator yang termasuk di dalamnya adalah penjumlahan (+), pengurangan (-), perkalian (*), pembagian (/), pembagian bulat (//), modulus atau sisa pembagian (%), dan perpangkatan (**). Operator aritmatika dapat digunakan pada berbagai tipe data numerik, seperti integer dan float. Penggunaannya banyak ditemukan dalam perhitungan nilai, transaksi, luas bangun, dan berbagai proses perhitungan lainnya.
2. Operator Perbandingan
Operator perbandingan digunakan untuk membandingkan dua nilai atau lebih. Hasil dari operasi perbandingan berupa nilai Boolean, yaitu True atau False. Operator yang digunakan meliputi sama dengan (==), tidak sama dengan (!=), lebih besar (>), lebih kecil (<), lebih besar atau sama dengan (>=), serta lebih kecil atau sama dengan (<=). Operator ini biasanya digunakan dalam percabangan dan pemeriksaan kondisi pada suatu program.
3. Operator Logika
Operator logika digunakan untuk menggabungkan atau memanipulasi beberapa kondisi. Python memiliki tiga operator logika utama, yaitu and, or, dan not. Operator and menghasilkan nilai benar apabila semua kondisi yang diperiksa bernilai benar, sedangkan or menghasilkan nilai benar apabila terdapat setidaknya satu kondisi yang benar. Sementara itu, not digunakan untuk membalik nilai Boolean dari suatu kondisi. Operator logika banyak digunakan dalam proses pengambilan keputusan pada program.
4. Operator Penugasan
Operator penugasan digunakan untuk memberikan nilai kepada variabel atau mengubah nilai yang telah tersimpan di dalam variabel. Operator yang paling dasar adalah =, sedangkan operator gabungan lainnya antara lain +=, -=, *=, /=, //=, %= dan **=. Operator tersebut dapat membuat penulisan program menjadi lebih sederhana, terutama ketika suatu variabel perlu diperbarui berdasarkan nilai sebelumnya.
5. Operator Bitwise
Operator bitwise digunakan untuk melakukan operasi pada tingkat bit dari suatu bilangan integer. Python menyediakan beberapa operator bitwise, yaitu AND (&), OR (|), XOR (^), NOT (~), left shift (<<), dan right shift (>>). Operasi ini bekerja dengan memproses representasi biner dari suatu bilangan. Operator bitwise biasanya digunakan dalam pengolahan data tingkat rendah, manipulasi bit, sistem komputer, dan beberapa kebutuhan pemrograman tertentu.
6. Operator Keanggotaan
Operator keanggotaan digunakan untuk memeriksa apakah suatu nilai terdapat di dalam sebuah objek seperti string, list, tuple, atau set. Python memiliki operator in dan not in. Operator in menghasilkan True apabila suatu nilai ditemukan di dalam objek, sedangkan not in menghasilkan True apabila nilai tersebut tidak ditemukan. Operator ini dapat digunakan untuk melakukan pemeriksaan data dengan cara yang sederhana.
7. Operator Identitas
Operator identitas digunakan untuk mengetahui apakah dua variabel mengacu pada objek yang sama di dalam memori. Python menyediakan operator is dan is not. Operator is menghasilkan True apabila kedua variabel mengacu pada objek yang sama, sedangkan is not menghasilkan True apabila keduanya tidak mengacu pada objek yang sama. Operator identitas berbeda dengan operator == yang digunakan untuk membandingkan nilai.

## 2.2 Kode Program

### 2.2.1 Program aritmatika_2611531021
</div>

In [1]:
angka1_1011=int(input("Input angka-1: "))
angka2_1011=int(input("input angka-2: "))

#penjumlahan
hasil1_1011= angka1_1011 + angka2_1011
print("\nOperator penjumlahan")
print("hasil =", hasil1_1011)

#pengurangan
hasil2_1011 = angka1_1011 - angka2_1011
print("\nOperator pengurangan")
print("hasil =", hasil2_1011)

#perkalian
hasil3_1011 = angka1_1011 * angka2_1011
print("\nOperator perkalian")
print("hasil =", hasil3_1011)

#pembagian, pembagian bulat, dan sisa bagi
if angka2_1011 != 0:
    hasil4_1011 = angka1_1011 / angka2_1011
    print("\nOperator pembagian")
    print("hasil =", hasil4_1011)

    hasil5_1011 = angka1_1011 // angka2_1011
    print("\nOperator pembagian bulat")
    print("hasil =", hasil5_1011)

    hasil6_1011 = angka1_1011 % angka2_1011
    print("\nOperator sisa bagi")
    print("hasil =", hasil6_1011)
else:
    print("Angka kedua tidak boleh bernilai 0.")

#pangkat
hasil7_1011 = angka1_1011 ** angka2_1011
print("\nOperator pangkat")
print("hasil =", hasil7_1011)


Operator penjumlahan
hasil = 8

Operator pengurangan
hasil = -2

Operator perkalian
hasil = 15

Operator pembagian
hasil = 0.6

Operator pembagian bulat
hasil = 0

Operator sisa bagi
hasil = 3

Operator pangkat
hasil = 243


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

**Penjelasan:**

**1. Memasukkan angka**<br>
angka1_1011 = int(input("Input angka-1: "))
angka2_1011 = int(input("Input angka-2: "))
Program meminta pengguna untuk memasukkan dua buah angka. Fungsi input() digunakan untuk menerima masukan dari pengguna, sedangkan int() digunakan untuk mengubah masukan tersebut menjadi tipe data integer.

**2. Operator Penjumlahan**<br>
hasil_1011 = angka1_1011 + angka2_1011
Operator + digunakan untuk menjumlahkan nilai dari angka1_1011 dan angka2_1011. Hasil dari operasi tersebut kemudian disimpan ke dalam variabel hasil_1011.

**3. Operator Pengurangan**<br>
hasil_1011 = angka1_1011 - angka2_1011
Operator - digunakan untuk melakukan pengurangan dengan cara mengurangi nilai angka1_1011 menggunakan nilai angka2_1011.

**4. Operator Perkalian**<br>
hasil_1011 = angka1_1011 * angka2_1011
Operator * digunakan untuk mengalikan nilai dari kedua angka yang telah dimasukkan oleh pengguna.

**5. Operator Pembagian**<br>
hasil_1011 = angka1_1011 / angka2_1011
Operator / digunakan untuk melakukan operasi pembagian antara angka1_1011 dan angka2_1011. Hasil dari operasi ini umumnya berupa tipe data float.

**6. Operator Pembagian Bulat**<br>
hasil_1011 = angka1_1011 // angka2_1011
Operator // digunakan untuk melakukan pembagian dengan mengambil hasil bilangan bulatnya saja, sehingga bagian desimal dari hasil pembagian tidak digunakan.

**7. Operator Sisa Bagi**<br>
hasil_1011 = angka1_1011 % angka2_1011
Operator % digunakan untuk memperoleh nilai sisa dari pembagian antara angka1_1011 dengan angka2_1011.

**8. Pengecekan Nilai Angka Kedua**<br>
if angka2_1011 != 0:
Program melakukan pengecekan untuk memastikan bahwa nilai angka2_1011 tidak sama dengan 0. Pengecekan ini diperlukan karena operasi pembagian dengan angka 0 tidak dapat dilakukan.

Jika angka2_1011 bernilai 0, maka program akan menampilkan pesan:
print("Angka kedua tidak boleh bernilai 0.")

**9. Operator Pangkat**<br>
hasil_1011 = angka1_1011 ** angka2_1011
Operator ** digunakan untuk melakukan operasi perpangkatan. Pada operasi ini, nilai angka1_1011akan dipangkatkan menggunakan nilai angka2_1011.


### 2.2.2 Program assigment_2611531011
</div>

In [2]:
angka1_1011 = int(input("Input angka-1: "))
angka2_1011 = int(input("Input angka-2: "))

print("\nnilai awal angka1_1011 =", angka1_1011)
print("nilai awal angka2_1011 =", angka2_1011)

#assignment biasa
hasil1_1011 = angka1_1011
print("\nAssignment biasa (=)")
print("hasil =", hasil1_1011)

#assignment penambahan
hasil2_1011 = angka1_1011
hasil2_1011 += angka2_1011
print("\nAssignment penambahan (+=)")
print("hasil =", hasil2_1011)

#assignment pengurangan
hasil3_1011 = angka1_1011
hasil3_1011 -= angka2_1011
print("\nAssignment pengurangan (-=)")
print("hasil =", hasil3_1011)

#assignment perkalian
hasil4_1011 = angka1_1011
hasil4_1011 *= angka2_1011
print("\nAssignment perkalian (*=)")
print("hasil =", hasil4_1011)

#assignment pembagian
if angka2_1011 != 0:
    hasil5_1011 = angka1_1011
    hasil5_1011 /= angka2_1011
    print("\nAssignment pembagian (/=)")
    print("hasil =", hasil5_1011)

    #assignment pembagian bulat
    hasil6_1011 = angka1_1011
    hasil6_1011 //= angka2_1011
    print("\nAssignment pembagian bulat (//=)")
    print("hasil =", hasil6_1011)

    #assignment sisa bagi
    hasil7_1011 = angka1_1011
    hasil7_1011 %= angka2_1011
    print("\nAssignment sisa bagi (%=)")
    print("hasil =", hasil7_1011)
else:
    print("\npembagian tidak dapat dilakukan")
    print("angka kedua tidak boleh bernilai 0.")

#operator tambahan: assignment pangkat
hasil8_1011 = angka1_1011
hasil8_1011 **= angka2_1011
print("\nAssignment pangkat (**=)")
print("hasil =", hasil8_1011)


nilai awal angka1_1011 = 3
nilai awal angka2_1011 = 5

Assignment biasa (=)
hasil = 3

Assignment penambahan (+=)
hasil = 8

Assignment pengurangan (-=)
hasil = -2

Assignment perkalian (*=)
hasil = 15

Assignment pembagian (/=)
hasil = 0.6

Assignment pembagian bulat (//=)
hasil = 0

Assignment sisa bagi (%=)
hasil = 3

Assignment pangkat (**=)
hasil = 243


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:

1. Memasukkan angka <br>
angka1_1011 = int(input("Input angka-1: "))
angka2_1011 = int(input("Input angka-2: "))
Program meminta pengguna untuk memasukkan dua buah angka. Fungsi input() digunakan untuk menerima masukan dari pengguna, kemudian int() mengubah masukan tersebut menjadi tipe data integer.

2. Menampilkan nilai awal <br>
print("\nNilai awal angka1 =", angka1_1011)
print("Nilai angka2 =", angka2_1011)
Bagian ini berfungsi untuk menampilkan nilai awal yang tersimpan pada variabel angka1_1011 dan angka2_1011.

3. Assignment biasa (=) <br>
hasil_1011 = angka1_1011
Operator = digunakan untuk menetapkan nilai angka1_1011 ke dalam variabel hasil_1011.

4. Assignment penambahan (+=) <br>
hasil_1011 = angka1_1011
hasil_1011 += angka2_1011
Operator += digunakan untuk menambahkan nilai angka2_1011 ke nilai yang terdapat pada hasil_1011.
Bentuk lengkapnya:
hasil_1011 = hasil_1011 + angka2_1011

5. Assignment pengurangan (-=) <br>
hasil_1011 = angka1_1011
hasil_1011 -= angka2_1011
Operator -= digunakan untuk mengurangi nilai hasil_1011 dengan angka2_1011.
Bentuk lengkapnya:
hasil_1011 = hasil_1011 - angka2_1011

6. Assignment perkalian (*=) <br>
hasil_1011 = angka1_1011
hasil_1011 *= angka2_1011
Operator *= digunakan untuk mengalikan nilai hasil_1011 dengan nilai angka2_1011.
Bentuk lengkapnya:
hasil_1011 = hasil_1011 * angka2_1011

7. Assignment pembagian (/=) <br>
hasil_1011 = angka1_1011
hasil_1011 /= angka2_1011
Operator /= digunakan untuk membagi nilai hasil_1011 dengan angka2_1011.
Bentuk lengkapnya:
hasil_1011 = hasil_1011 / angka2_1011
Hasil operasi pembagian dengan operator / biasanya memiliki tipe data float.

8. Assignment pembagian bulat (//=) <br>
hasil_1011 = angka1_1011
hasil_1011 //= angka2_1011
Operator //= digunakan untuk melakukan pembagian bulat dan menyimpan hasilnya kembali ke dalam variabel hasil_1011.
Bentuk lengkapnya:
hasil_1011 = hasil_1011 // angka2_1011

9. Assignment sisa bagi (%=) <br>
hasil_1011 = angka1_1011
hasil_1011 %= angka2_1011
Operator %= digunakan untuk mencari sisa dari pembagian dan menyimpan hasilnya pada hasil_1011.
Bentuk lengkapnya:
hasil_1011 = hasil_1011 % angka2_1011

10. Pengecekan nilai 0 <br>
if angka2_1011 != 0:
Program mengecek apakah nilai angka2_1011 bukan 0. Pengecekan ini dilakukan karena operasi pembagian, pembagian bulat, dan sisa bagi tidak dapat menggunakan 0 sebagai pembagi.

Jika nilai angka2_1011 adalah 0, program akan menampilkan:
print("\nPembagian tidak dapat dilakukan.")
print("Angka kedua tidak boleh bernilai 0.")

11. Assignment perpangkatan (=)** <br>
hasil_1011 = angka1_1011
hasil_1011 **= angka2_1011
Operator **= digunakan untuk melakukan operasi perpangkatan sekaligus menyimpan hasilnya ke dalam variabel hasil_1011.
Bentuk lengkapnya:
hasil_1011 = hasil_1011 ** angka2_1011
### 2.2.3 Program perbandingan_2611531011
</div>

In [3]:
angka1_1011 = int(input("Input angka-1: "))
angka2_1011 = int(input("Input angka-2: "))

#lebih besar dari
hasil1_1011 = angka1_1011 > angka2_1011
print("\nOperator lebih besar dari")
print("angka-1 > angka-2 =", hasil1_1011)

#lebih kecil dari
hasil2_1011 = angka1_1011 < angka2_1011
print("\nOperator lebih kecil dari")
print("angka-1 < angka-2 =", hasil2_1011)

#lebih besar dari atau sama dengan
hasil3_1011 = angka1_1011 >= angka2_1011
print("\nOperator lebih besar dari atau sama dengan")
print("angka-1 >= angka-2 =", hasil3_1011)

#lebih kecil dari atau sama dengan
hasil4_1011 = angka1_1011 <= angka2_1011
print("\nOperator lebih kecil dari atau sama dengan")
print("angka-1 <= angka-2 =", hasil4_1011)

#sama dengan
hasil5_1011 = angka1_1011 == angka2_1011
print("\nOperator sama dengan")
print("angka-1 == angka-2 =", hasil5_1011)

#tidak sama dengan
hasil6_1011 = angka1_1011 != angka2_1011
print("\nOperator tidak sama dengan")
print("angka-1 != angka-2 =", hasil6_1011)

#perbandingan berantai 
hasil7_1011 = 0 < angka1_1011 < 100
print("\nOperator perbandingan berantai")
print("0 < angka-1 < 100 =", hasil7_1011)

hasil8_1011 = 0 < angka2_1011 < 100
print("0 < angka-2 < 100 =", hasil8_1011)
angka1_1011 = int(input("Input angka-1: "))
angka2_1011 = int(input("Input angka-2: "))

#lebih besar dari
hasil1_1011 = angka1_1011 > angka2_1011
print("\nOperator lebih besar dari")
print("angka-1 > angka-2 =", hasil1_1011)

#lebih kecil dari
hasil2_1011 = angka1_1011 < angka2_1011
print("\nOperator lebih kecil dari")
print("angka-1 < angka-2 =", hasil2_1011)

#lebih besar dari atau sama dengan
hasil3_1011 = angka1_1011 >= angka2_1011
print("\nOperator lebih besar dari atau sama dengan")
print("angka-1 >= angka-2 =", hasil3_1011)

#lebih kecil dari atau sama dengan
hasil4_1011 = angka1_1011 <= angka2_1011
print("\nOperator lebih kecil dari atau sama dengan")
print("angka-1 <= angka-2 =", hasil4_1011)

#sama dengan
hasil5_1011 = angka1_1011 == angka2_1011
print("\nOperator sama dengan")
print("angka-1 == angka-2 =", hasil5_1011)

#tidak sama dengan
hasil6_1011 = angka1_1011 != angka2_1011
print("\nOperator tidak sama dengan")
print("angka-1 != angka-2 =", hasil6_1011)

#perbandingan berantai 
hasil7_1011 = 0 < angka1_1011 < 100
print("\nOperator perbandingan berantai")
print("0 < angka-1 < 100 =", hasil7_1011)

hasil8_1011 = 0 < angka2_1011 < 100
print("0 < angka-2 < 100 =", hasil8_1011)



Operator lebih besar dari
angka-1 > angka-2 = False

Operator lebih kecil dari
angka-1 < angka-2 = True

Operator lebih besar dari atau sama dengan
angka-1 >= angka-2 = False

Operator lebih kecil dari atau sama dengan
angka-1 <= angka-2 = True

Operator sama dengan
angka-1 == angka-2 = False

Operator tidak sama dengan
angka-1 != angka-2 = True

Operator perbandingan berantai
0 < angka-1 < 100 = True
0 < angka-2 < 100 = True

Operator lebih besar dari
angka-1 > angka-2 = False

Operator lebih kecil dari
angka-1 < angka-2 = True

Operator lebih besar dari atau sama dengan
angka-1 >= angka-2 = False

Operator lebih kecil dari atau sama dengan
angka-1 <= angka-2 = True

Operator sama dengan
angka-1 == angka-2 = False

Operator tidak sama dengan
angka-1 != angka-2 = True

Operator perbandingan berantai
0 < angka-1 < 100 = True
0 < angka-2 < 100 = True


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:

1. Memasukkan angka
angka1_1011 = int(input("Input angka-1: "))
angka2_1011 = int(input("Input angka-2: "))
Program meminta pengguna untuk memasukkan dua angka. Fungsi input() digunakan untuk menerima masukan, kemudian int() mengubah masukan tersebut menjadi tipe data integer.

2. Operator lebih besar dari (>) <br>
hasil_1011 = angka1_1011 > angka2_1011
Operator > digunakan untuk memeriksa apakah nilai angka1_1011 lebih besar dibandingkan angka2_1011. Hasil perbandingan berupa nilai Boolean, yaitu True atau False.

3. Operator lebih kecil dari (<) <br>
hasil_1011 = angka1_1011 < angka2_1011
Operator < digunakan untuk mengetahui apakah nilai angka1_1011 lebih kecil daripada nilai angka2_1011.

4. Operator lebih besar dari atau sama dengan (>=) <br>
hasil_1011 = angka1_1011 >= angka2_1011
Operator >= digunakan untuk memeriksa apakah angka1_1011 memiliki nilai yang lebih besar atau sama dengan angka2_1011.

5. Operator lebih kecil dari atau sama dengan (<=) <br>
hasil_1011 = angka1_1011 <= angka2_1011
Operator <= digunakan untuk mengecek apakah nilai angka1_1011 lebih kecil atau sama dengan nilai angka2_1011.

6. Operator sama dengan (==) <br>
hasil_1011 = angka1_1011 == angka2_1011
Operator == digunakan untuk membandingkan apakah angka1_1011 dan angka2_1011 mempunyai nilai yang sama.
Perlu dibedakan antara = dan ==.
Operator = digunakan untuk memberikan atau menetapkan nilai, sedangkan == digunakan untuk membandingkan dua nilai.

7. Operator tidak sama dengan (!=) <br>
hasil_1011 = angka1_1011 != angka2_1011
Operator != digunakan untuk memeriksa apakah nilai angka1_1011 berbeda dengan nilai angka2_1011.

8. Perbandingan berantai untuk angka pertama <br>
hasil_1011 = 0 < angka1_1011 < 100
Kode tersebut menggunakan perbandingan berantai (chained comparison). Artinya, program memeriksa apakah nilai angka1_1011 berada di antara 0 dan 100, dengan dua kondisi berikut:
0 < angka1_1011 dan angka1_1011 < 100
Jika kedua kondisi terpenuhi, maka hasil perbandingannya adalah True.

9. Perbandingan berantai untuk angka kedua <br>
hasil_1011 = 0 < angka2_1011 < 100
Bagian ini menggunakan cara yang sama seperti sebelumnya, tetapi diterapkan pada angka2_1011. Program akan memeriksa apakah nilai angka2_1011 berada di antara 0 dan 100.

### 2.2.4 Program logika_2611531011
</div>

In [6]:
#memasukkan nilai boolean
#input tidak peka terhadap huruf besar dan kecil
a1_1011 = input("Input nilai boolean-1 (True/False): ").strip().lower() == "true"
a2_1011 = input("Input nilai boolean-2 (True/False): ").strip().lower() == "true"

print("\nA1 =", a1_1011)
print("A2 =", a2_1011)

#konjungsi: bernilai true jika keduanya true
hasil1_1011 = a1_1011 and a2_1011
print("\nKonjungsi (AND)")
print("A1 AND A2 =", hasil1_1011)

#disjungsi: bernilai true jika salah satu true
hasil2_1011 = a1_1011 or a2_1011
print("\nDisjungsi (OR)")
print("A1 OR A2 =", hasil2_1011)

#negasi: A1: membalik nilai A1
hasil3_1011 = not a2_1011
print("\nNegasi (NOT)")
print("NOT A2 =", hasil3_1011)

#xor: bernilai true jika kedua nilai berbeda
hasil4_1011 = a1_1011 != a2_1011
print("\ndiskonjungsi eksklusif (XOR)")
print("A1 XOR A2 =", hasil4_1011) 



A1 = True
A2 = False

Konjungsi (AND)
A1 AND A2 = False

Disjungsi (OR)
A1 OR A2 = True

Negasi (NOT)
NOT A2 = True

diskonjungsi eksklusif (XOR)
A1 XOR A2 = True


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:

1. Memasukkan nilai Boolean <br>
a1_1011 = input("Input nilai boolean-1 (true/false): ").strip().lower() == "true"
a2_1011 = input("Input nilai boolean-2 (true/false): ").strip().lower() == "true"
Program meminta pengguna untuk memasukkan nilai true atau false.

input() digunakan untuk menerima masukan dari pengguna.
.strip() berfungsi menghapus spasi yang terdapat di awal dan akhir masukan.
.lower() mengubah semua huruf menjadi huruf kecil sehingga TRUE, True, dan true dapat diproses dengan cara yang sama.
== "true" membandingkan masukan dengan teks "true" dan menghasilkan nilai Boolean berupa True atau False.

2. Menampilkan nilai Boolean <br>
print("\nA1 = ", a1_1011)
print("A2 = ", a2_1011)
Bagian ini digunakan untuk menampilkan nilai Boolean dari a1_1011 dan a2_1011 setelah masukan diproses oleh program.

3. Konjungsi (AND) <br>
hasil_1011 = a1_1011 and a2_1011
Operator and digunakan untuk menggabungkan dua nilai atau kondisi. Hasilnya akan bernilai True apabila kedua nilai sama-sama True.

Contoh:
A1 = True
A2 = True
Hasil = True

Apabila salah satu atau kedua nilai bernilai False, maka hasilnya adalah False.

4. Disjungsi (OR) <br>
hasil_1011 = a1_1011 or a2_1011
Operator or akan menghasilkan True apabila setidaknya salah satu dari kedua nilai bernilai True. Hasilnya akan False hanya jika kedua nilai bernilai False.

5. Negasi A1 (NOT) <br>
hasil_1011 = not a1_1011
Operator not digunakan untuk membalik nilai Boolean pada a1_1011.
Jika A1 = True, maka not A1 = False.
Jika A1 = False, maka not A1 = True.

6. Negasi A2 (NOT) <br>
hasil_1011 = not a2_1011
Bagian ini berfungsi membalik nilai Boolean dari a2_1011. Cara kerjanya sama seperti operator not pada a1_1011.

7. Disjungsi Eksklusif (XOR) <br>
hasil_1011 = a1_1011 != a2_1011
Operator XOR digunakan untuk menghasilkan nilai True apabila kedua nilai Boolean berbeda. Jika kedua nilai sama, maka hasilnya adalah False.

### 2.2.5 Program lainnya_2611531011
</div>

In [10]:
#program operator keanggotaan dan identitas
print("==================================")
print("1. Operator Keanggotaan")
print("==================================")

#input beberapa data yang dipisahkan dengan koma
input_data_1011 = input("Input beberapa angka, pisahkan dengan koma: ")

#mengubah input menjadi list integer
data_1011 = [int(angka.strip()) for angka in input_data_1011.split(",")]

nilai_dicari_1011 = int(input("Input nilai yang dicari: "))

#operator in
hasil1_1011 = nilai_dicari_1011 in data_1011
print("\nOperator in")
print(nilai_dicari_1011, "in", data_1011, "=", hasil1_1011)

#operator not in
hasil2_1011 = nilai_dicari_1011 not in data_1011
print("\nOperator not in")
print(nilai_dicari_1011, "not in", data_1011, "=", hasil2_1011)

print("\n==================================")
print("2. Operator Identitas")
print("==================================")

#objek1 menggunakan list dari input pengguna
objek1_1011 = data_1011

#objek2 merujuk pada objek yang sama dengan objek1
objek2_1011 = objek1_1011

#objek3 memiliki isi sama, tetapi merupakan objek yang baru
objek3_1011 = data_1011.copy()

print("objek1 =", objek1_1011)
print("objek2 =", objek2_1011)
print("objek3 =", objek3_1011)

#operator is
hasil3_1011 = objek1_1011 is objek2_1011
print("\nOperator is")
print("objek1 is objek2 =", hasil3_1011)

#operator is not
hasil4_1011 = objek1_1011 is not objek3_1011
print("\nOperator is not")
print("objek1 is not objek3 =", hasil4_1011)

#membandingkan identitas dan nilai
print("\nperbandingkan identitas dan nilai")
print("objek1 is objek3 =", objek1_1011 is objek3_1011)
print("objek1 == objek3 =", objek1_1011 == objek3_1011)
#program operator keanggotaan dan identitas
print("==================================")
print("1. Operator Keanggotaan")
print("==================================")

#input beberapa data yang dipisahkan dengan koma
input_data_1011 = input("Input beberapa angka, pisahkan dengan koma: ")

#mengubah input menjadi list integer
data_1011 = [int(angka.strip()) for angka in input_data_1011.split(",")]

nilai_dicari_1011 = int(input("Input nilai yang dicari: "))

#operator in
hasil1_1011 = nilai_dicari_1011 in data_1011
print("\nOperator in")
print(nilai_dicari_1011, "in", data_1011, "=", hasil1_1011)

#operator not in
hasil2_1011 = nilai_dicari_1011 not in data_1011
print("\nOperator not in")
print(nilai_dicari_1011, "not in", data_1011, "=", hasil2_1011)

print("\n==================================")
print("2. Operator Identitas")
print("==================================")

#objek1 menggunakan list dari input pengguna
objek1_1011 = data_1011

#objek2 merujuk pada objek yang sama dengan objek1
objek2_1011 = objek1_1011

#objek3 memiliki isi sama, tetapi merupakan objek yang baru
objek3_1011 = data_1011.copy()

print("objek1 =", objek1_1011)
print("objek2 =", objek2_1011)
print("objek3 =", objek3_1011)

#operator is
hasil3_1011 = objek1_1011 is objek2_1011
print("\nOperator is")
print("objek1 is objek2 =", hasil3_1011)

#operator is not
hasil4_1011 = objek1_1011 is not objek3_1011
print("\nOperator is not")
print("objek1 is not objek3 =", hasil4_1011)

#membandingkan identitas dan nilai
print("\nperbandingkan identitas dan nilai")
print("objek1 is objek3 =", objek1_1011 is objek3_1011)
print("objek1 == objek3 =", objek1_1011 == objek3_1011)


1. Operator Keanggotaan

Operator in
6 in [1, 5, 6, 7] = True

Operator not in
6 not in [1, 5, 6, 7] = False

2. Operator Identitas
objek1 = [1, 5, 6, 7]
objek2 = [1, 5, 6, 7]
objek3 = [1, 5, 6, 7]

Operator is
objek1 is objek2 = True

Operator is not
objek1 is not objek3 = True

perbandingkan identitas dan nilai
objek1 is objek3 = False
objek1 == objek3 = True
1. Operator Keanggotaan

Operator in
7 in [1, 5, 6, 7] = True

Operator not in
7 not in [1, 5, 6, 7] = False

2. Operator Identitas
objek1 = [1, 5, 6, 7]
objek2 = [1, 5, 6, 7]
objek3 = [1, 5, 6, 7]

Operator is
objek1 is objek2 = True

Operator is not
objek1 is not objek3 = True

perbandingkan identitas dan nilai
objek1 is objek3 = False
objek1 == objek3 = True


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:


1. Menerima input beberapa angka
input_data_1011 = input("Masukkan beberapa angka, pisahkan dengan koma: ")
Program meminta pengguna memasukkan beberapa angka yang dipisahkan menggunakan tanda koma. Data yang dimasukkan pertama kali akan disimpan dalam bentuk string.

Contoh:
10, 20, 30, 40

2. Mengubah input menjadi list integer
data_1011 = [int(angka.strip()) for angka in input_data_1011.split(',')]
Kode ini digunakan untuk mengubah data input yang masih berupa string menjadi sebuah list dengan tipe data integer.

.split(',') memisahkan data berdasarkan tanda koma.
.strip() menghapus spasi yang terdapat pada setiap data.
int() mengubah data menjadi bilangan integer.
[ ... for ... ] digunakan untuk membentuk list.

Contoh:
Input: 10, 20, 30
Hasil: [10, 20, 30]

3. Memasukkan nilai yang ingin dicari
nilai_dicari_1011 = int(input("Masukkan angka yang ingin dicari: "))
Program meminta pengguna memasukkan angka yang akan diperiksa keberadaannya di dalam data_1011. Nilai yang dimasukkan kemudian dikonversi menjadi tipe data integer.

4. Operator keanggotaan in
hasil_1011 = nilai_dicari_1011 in data_1011
Operator in digunakan untuk memeriksa apakah suatu nilai terdapat di dalam sebuah list. Jika nilai ditemukan, hasilnya True, sedangkan jika tidak ditemukan hasilnya False.

Contoh:
data = [10, 20, 30]
20 in data → True
50 in data → False

5. Operator keanggotaan not in
hasil_1011 = nilai_dicari_1011 not in data_1011
Operator not in digunakan untuk mengecek apakah suatu nilai tidak terdapat dalam list. Jika nilai tidak ditemukan, hasilnya True. Sebaliknya, jika nilai ditemukan, hasilnya False.

6. Membuat object1
object1_1011 = data_1011
Variabel object1_1011 dibuat dengan menunjuk ke objek list yang sama dengan data_1011. Dengan demikian, keduanya merujuk pada objek yang sama.

7. Membuat object2
object2_1011 = object1_1011
Variabel object2_1011 menunjuk ke objek yang sama dengan object1_1011. Artinya, kedua variabel tersebut memiliki identitas objek yang sama.

8. Membuat object3
object3_1011 = data_1011.copy()
Fungsi copy() digunakan untuk membuat objek list baru dengan isi yang sama seperti data_1011. Oleh karena itu, isi object3_1011 sama dengan objek sebelumnya, tetapi identitas objeknya berbeda.

9. Menampilkan isi objek
print("object1 =",object1_1011)
print("object2 =",object2_1011)
print("object3 =",object3_1011)
Bagian ini digunakan untuk menampilkan isi dari object1_1011, object2_1011, dan object3_1011. Karena ketiga objek memiliki isi yang sama, nilai yang ditampilkan akan terlihat sama.

10. Operator identitas is
hasil_1011 = object1_1011 is object2_1011
Operator is digunakan untuk memeriksa apakah dua variabel merujuk pada objek yang sama. Karena object2_1011 dibuat dari object1_1011, keduanya menunjuk objek yang sama sehingga hasilnya adalah True.

11. Operator identitas is not
hasil_1011 = object1_1011 is not object2_1011
Operator is not digunakan untuk mengecek apakah dua variabel merujuk pada objek yang berbeda. Karena object1_1011 dan object2_1011 menunjuk objek yang sama, hasilnya adalah False.

12. Membandingkan identitas dan nilai
print("objek1 is objek3 =",object1_1011 is object3_1011)
print("objek1 == objek3 =",object1_1011 == object3_1011)
Baris pertama menggunakan operator is untuk membandingkan identitas objek. Karena object3_1011 dibuat menggunakan .copy(), objek tersebut memiliki identitas yang berbeda sehingga hasilnya False.

Baris kedua menggunakan operator == untuk membandingkan isi atau nilai dari kedua objek. Karena isi object1_1011 dan object3_1011 sama, hasil perbandingannya adalah True.

### 2.2.6 Program bitwise_2611531011
</div>

In [12]:
#program  ini menggunakan fungsi input ()

print("\n==================================")
print("3. Operator Bitwise")
print("==================================")

angka1_1011 = int(input("masukkan angka bitwise-1: "))
angka2_1011 = int(input("masukkan angka bitwise-2: "))

print("\nangka dalam bentuk desiman dan biner")
print("angka1 =", angka1_1011, "| biner =", bin(angka1_1011))
print("angka2 =", angka2_1011, "| biner =", bin(angka2_1011))

#bitwise AND
hasil1_1011 = angka1_1011 & angka2_1011
print("\nBitwise AND (&)")
print(angka1_1011, "&", angka2_1011, "=", hasil1_1011)
print("biner hasil =", bin(hasil1_1011))
print("biner hasil (8 bit) =", format(hasil1_1011, '08b'))

#bitwise OR
hasil2_1011 = angka1_1011 | angka2_1011
print("\nBitwise OR (|)")
print(angka1_1011, "|", angka2_1011, "=", hasil2_1011)
print("biner hasil =", bin(hasil2_1011))
print("biner hasil (8 bit) =", format(hasil2_1011, '08b'))

#bitwise XOR
hasil3_1011 = angka1_1011 ^ angka2_1011
print("\nBitwise XOR (^)")
print(angka1_1011, "^", angka2_1011, "=", hasil3_1011)
print("biner hasil =", bin(hasil3_1011))
print("biner hasil (8 bit) =", format(hasil3_1011, '08b'))

#bitwise NOT
hasil4_1011 = ~angka1_1011
print("\nBitwise NOT (~)")
print("~", angka1_1011, "=", hasil4_1011)
print("biner hasil =", bin(hasil4_1011))
print("biner hasil (8 bit) =", format(hasil4_1011, '08b'))

#bitwise geser kiri
jumlah_geser = int(input("\nmasukkan jumlah pergeseran bit: "))

hasil5_1011 = angka1_1011 << jumlah_geser
print("\nBitwise Geser Kiri (<<)")
print(angka1_1011, "<<", jumlah_geser, "=", hasil5_1011)
print("biner hasil =", bin(hasil5_1011))
print("biner hasil (8 bit) =", format(hasil5_1011, '08b'))

#bitwise geser kanan
hasil6_1011 = angka1_1011 >> jumlah_geser
print("\nBitwise Geser Kanan (>>)")
print(angka1_1011, ">>", jumlah_geser, "=", hasil6_1011)
print("biner hasil =", bin(hasil6_1011))
print("biner hasil (8 bit) =", format(hasil6_1011, '08b'))


3. Operator Bitwise

angka dalam bentuk desiman dan biner
angka1 = 10 | biner = 0b1010
angka2 = 20 | biner = 0b10100

Bitwise AND (&)
10 & 20 = 0
biner hasil = 0b0
biner hasil (8 bit) = 00000000

Bitwise OR (|)
10 | 20 = 30
biner hasil = 0b11110
biner hasil (8 bit) = 00011110

Bitwise XOR (^)
10 ^ 20 = 30
biner hasil = 0b11110
biner hasil (8 bit) = 00011110

Bitwise NOT (~)
~ 10 = -11
biner hasil = -0b1011
biner hasil (8 bit) = -0001011

Bitwise Geser Kiri (<<)
10 << 1 = 20
biner hasil = 0b10100
biner hasil (8 bit) = 00010100

Bitwise Geser Kanan (>>)
10 >> 1 = 5
biner hasil = 0b101
biner hasil (8 bit) = 00000101


<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Penjelasan:

1. Memasukkan dua angka <br>
angka1_1011 = int(input("Masukkan angka bitwise-1: "))
angka2_1011 = int(input("Masukkan angka bitwise-2: "))
Program meminta pengguna untuk memasukkan dua buah angka. Fungsi input() digunakan untuk menerima masukan, sedangkan int() mengubah masukan tersebut menjadi tipe data integer.

2. Menampilkan angka desimal dan biner <br>
print("\nAngka dalam bentuk desimal dan biner")
print("angka1 =",angka1_1011,"| biner =",bin(angka1_1011))
print("angka1 =",angka2_1011,"| biner =",bin(angka2_1011))
Bagian ini menampilkan nilai kedua angka dalam bentuk desimal dan biner. Fungsi bin() digunakan untuk mengubah bilangan integer menjadi representasi biner.

3. Operasi AND (&) <br>
hasil_1011 = angka1_1011 & angka2_1011
Operator & digunakan untuk melakukan operasi AND pada setiap bit dari kedua angka. Suatu bit akan menghasilkan 1 apabila kedua bit yang dibandingkan sama-sama bernilai 1.

print(angka1_1011,"&",angka2_1011,hasil_1011)
print("Biner hasil =",bin(hasil_1011))
print("Biner hasil (8 bit) =",format(hasil_1011,"08b"))

Perintah tersebut menampilkan hasil operasi dalam bentuk desimal, biner, serta biner dengan panjang 8 bit. Fungsi format(..., "08b") digunakan untuk menampilkan hasil biner dengan minimal 8 digit.

4. Operasi OR (|) <br>
hasil_1011 = angka1_1011 | angka2_1011
Operator | digunakan untuk melakukan operasi OR pada setiap bit. Hasil bit akan bernilai 1 apabila salah satu atau kedua bit bernilai 1.

print(angka1_1011,"|",angka2_1011,hasil_1011)
print("Biner hasil =",bin(hasil_1011))
print("Biner hasil (8 bit) =",format(hasil_1011,"08b"))

Bagian ini digunakan untuk menampilkan hasil operasi OR dalam bentuk desimal dan biner.

5. Operasi XOR (^) <br>
hasil_1011 = angka1_1011 ^ angka2_1011
Operator ^ digunakan untuk melakukan operasi XOR. Operator ini menghasilkan bit 1 apabila kedua bit yang dibandingkan memiliki nilai yang berbeda.

Contoh:
0 ^ 0 = 0
0 ^ 1 = 1
1 ^ 0 = 1
1 ^ 1 = 0

Perintah print() selanjutnya digunakan untuk menampilkan hasil operasi XOR dalam bentuk desimal dan biner.

6. Operasi NOT (~) <br>
hasil_1011 = ~angka1_1011
Operator ~ digunakan untuk membalik nilai setiap bit pada angka1_1011. Bit yang bernilai 1 akan dibalik menjadi 0, sedangkan bit 0 menjadi 1.

print(angka1_1011,"~",angka2_1011,hasil_1011)
Perintah tersebut digunakan untuk menampilkan hasil operasi NOT. Namun, angka2_1011 tetap ditampilkan meskipun operasi NOT hanya dilakukan pada angka1_1011. Jika ingin menampilkan hasil yang lebih sesuai dengan operasinya, dapat menggunakan:
print("~", angka1_1011, "=", hasil_1011)

7. Memasukkan jumlah pergeseran bit <br>
jumlah_geser_1011 = int(input("\nMasukkan jumlah pergeseran bit: "))
Program meminta pengguna menentukan jumlah posisi bit yang akan digeser. Masukan tersebut kemudian diubah menjadi tipe data integer menggunakan int().

8. Pergeseran bit ke kiri (<<) <br>
hasil_1011 = angka1_1011 << jumlah_geser_1011
Operator << digunakan untuk menggeser bit angka1_1011 ke arah kiri sebanyak jumlah_geser_1011 posisi.

Secara sederhana, pergeseran satu posisi ke kiri pada bilangan positif umumnya setara dengan mengalikan bilangan tersebut dengan 2.

Contoh:
5 = 0101
5 << 1 = 1010 = 10

9. Menampilkan hasil geser kiri <br>
print("\nBitwise geser kiri (<<)")
print(angka1_1011,"<<",jumlah_geser_1011,"=",hasil_1011)
print("Biner hasil =",bin(hasil_1011))
print("Biner hasil (8 bit) =",format(hasil_1011,"08b"))

Bagian ini menampilkan hasil pergeseran bit ke kiri dalam bentuk desimal, biner, dan biner dengan panjang 8 bit.

10. Pergeseran bit ke kanan (>>) <br>
hasil_1011 = angka1_1011 >> jumlah_geser_1011
Operator >> digunakan untuk menggeser bit angka1_1011 ke arah kanan sebanyak jumlah posisi yang ditentukan.

Untuk bilangan positif, pergeseran satu posisi ke kanan secara sederhana biasanya setara dengan membagi bilangan dengan 2 dan mengambil hasil bilangan bulatnya.

Contoh:
10 = 1010
10 >> 1 = 0101 = 5

11. Menampilkan hasil geser kanan <br>
print("\nBitwise geser kiri (>>)")
print(angka1_1011,">>",jumlah_geser_1011,"=",hasil_1011)
print("Biner hasil =",bin(hasil_1011))
print("Biner hasil (8 bit) =",format(hasil_1011,"08b"))

Perintah tersebut digunakan untuk menampilkan hasil pergeseran bit ke kanan dalam bentuk desimal, biner, dan biner dengan panjang 8 bit.

---
## Tugas Praktikum

**Soal :**

<div align="center">
<img src="image/soal tugas 3 (1).png" width="500"><br>
<img src="image/soal tugas 3 (2).png" width="500"><br>
<em>Gambar 2.2 : Soal Tugas Praktikum</em>
</div>

**Kode Jawaban :**

a. Kode Program tugas3_2611531011
</div>

In [13]:
# ==============================================================================
# PROGRAM SIMULASI TRANSAKSI DAN VALIDASI AKSES TOKO
# ==============================================================================

# 1. INPUT DATA UTAMA
print("=== SISTEM TRANSAKSI TOKO ===")
nama_1011 = input("Masukkan Nama Pelanggan : ")
status_1011 = input("Masukkan Status Pelanggan (member/nonmember) : ")
total_belanja_1011 = int(input("Masukkan Total Belanja : "))
jumlah_barang_1011 = int(input("Masukkan Jumlah Barang : "))
kode_promo_1011 = input("Masukkan Kode Promo : ")

# List Promo yang Diterima
daftar_promo_1011 = ["SABTUMAJU", "MINGGUSERU", "LAPANDELAPAN", "HEMATBERSAMA"]

# 2. EVALUASI KONDISI (OPERATOR PERBANDINGAN & KEANGGOTAAN)
is_member_1011 = (status_1011.lower() == "member")
is_cukup_belanja_1011 = total_belanja_1011 >= 200000
is_cukup_barang_1011 = jumlah_barang_1011 >= 3
is_promo_tersedia_1011 = kode_promo_1011 in daftar_promo_1011

# Evaluasi Logika Kombinasi
dapat_diskon_1011 = is_cukup_belanja_1011 or is_cukup_barang_1011
dapat_promo_1011 = is_promo_tersedia_1011

# 3. KALKULASI ARITMATIKA & AUGMENTED ASSIGNMENT
persen_diskon_1011 = 0.05
potongan_harga_1011 = total_belanja_1011 * persen_diskon_1011

# Menggunakan operator penugasan (-=) untuk total bayar
total_pembayaran_1011 = float(total_belanja_1011)
total_pembayaran_1011 -= potongan_harga_1011

rata_rata_harga_1011 = total_pembayaran_1011 / jumlah_barang_1011

# 4. OPERASI BITWISE
# Pembuatan Mask Bitwise
BIT_MEMBER_1011 = 1 if is_member_1011 else 0
BIT_BELANJA_1011 = 2 if is_cukup_belanja_1011 else 0
BIT_BARANG_1011 = 4 if is_cukup_barang_1011 else 0
BIT_PROMO_1011 = 8 if is_promo_tersedia_1011 else 0

# Menggabungkan Flag Status dengan Bitwise OR (|)
kode_transaksi_1011 = BIT_MEMBER_1011 | BIT_BELANJA_1011 | BIT_BARANG_1011 | BIT_PROMO_1011
kode_referensi_1011 = BIT_MEMBER_1011 | BIT_BELANJA_1011 | BIT_PROMO_1011

# Operasi Bitwise Spesifik
cek_member_1011 = kode_transaksi_1011 & 0b0001
cek_promo_1011 = kode_transaksi_1011 & 0b1000
hasil_xor_1011 = kode_transaksi_1011 ^ kode_referensi_1011
hasil_shift_1011 = kode_transaksi_1011 << 1

# ==============================================================================
# CETAK OUTPUT TRANSAKSI
# ==============================================================================

print("\n=== DATA TRANSAKSI ===")
print(f"Nama Pelanggan       : {nama_1011}")
print(f"Status Pelanggan     : {status_1011}")
print(f"Total Belanja        : Rp{total_belanja_1011}")
print(f"Jumlah Barang        : {jumlah_barang_1011}")
print(f"Kode Promo           : {kode_promo_1011}")

print("\n=== HASIL VALIDASI ===")
print(f"Belanja >= Rp200000        : {is_cukup_belanja_1011}")
print(f"Jumlah Barang >= 3         : {is_cukup_barang_1011}")
print(f"Status Member              : {is_member_1011}")
print(f"Kode Promo Tersedia        : {is_promo_tersedia_1011}")
print(f"Mendapatkan Diskon         : {dapat_diskon_1011}")
print(f"Mendapatkan Promo          : {dapat_promo_1011}")

print("\n=== HASIL PERHITUNGAN ===")
print(f"Diskon                      : Rp{potongan_harga_1011:.1f}")
print(f"Total Pembayaran            : Rp{total_pembayaran_1011:.1f}")
print(f"Rata-rata Harga Barang      : Rp{rata_rata_harga_1011:.1f}")

print("\n=== HAK AKSES PELANGGAN ===")
print(f"Kode Hak Akses              : {bin(kode_transaksi_1011)[2:].zfill(4)}")
print(f"Member Access               : {is_member_1011}")
print(f"Promo Access                : {is_promo_tersedia_1011}")
print(f"Free Shipping Access        : {kode_promo_1011 == 'SABTUMAJU'}")

print("\n=== OPERASI BITWISE ===")
print("=== Kode Status Transaksi ===")
print(f"{BIT_MEMBER_1011:04b} | {BIT_BELANJA_1011:04b} | {BIT_BARANG_1011:04b} | {BIT_PROMO_1011:04b}")
print(f"Kode Biner   : {kode_transaksi_1011:04b}")
print(f"Kode Desimal : {kode_transaksi_1011}")

print("\n=== Pemeriksaan Status ===")
print("Cek Member")
print(f"{kode_transaksi_1011:04b} & 0001")
print(f"Hasil Biner   : {cek_member_1011:04b}")
print(f"Hasil Desimal : {cek_member_1011}")

print("\nCek Promo")
print(f"{kode_transaksi_1011:04b} & 1000")
print(f"Hasil Biner   : {cek_promo_1011:04b}")
print(f"Hasil Desimal : {cek_promo_1011}")

print("\n=== Perbandingan Status ===")
print(f"Kode Transaksi : {kode_transaksi_1011:04b}")
print(f"Kode Referensi : {kode_referensi_1011:04b}")
print(f"{kode_transaksi_1011:04b} ^ {kode_referensi_1011:04b}")
print(f"Hasil Biner   : {hasil_xor_1011:04b}")
print(f"Hasil Desimal : {hasil_xor_1011}")

print("\n=== Shift ===")
print(f"{kode_transaksi_1011:04b} << 1")
print(f"Hasil Biner   : {hasil_shift_1011:05b}")
print(f"Hasil Desimal : {hasil_shift_1011}")

print("=== SELESAI ===")

=== SISTEM TRANSAKSI TOKO ===

=== DATA TRANSAKSI ===
Nama Pelanggan       : yazid
Status Pelanggan     : member
Total Belanja        : Rp250000
Jumlah Barang        : 5
Kode Promo           : SABTUMAJU

=== HASIL VALIDASI ===
Belanja >= Rp200000        : True
Jumlah Barang >= 3         : True
Status Member              : True
Kode Promo Tersedia        : True
Mendapatkan Diskon         : True
Mendapatkan Promo          : True

=== HASIL PERHITUNGAN ===
Diskon                      : Rp12500.0
Total Pembayaran            : Rp237500.0
Rata-rata Harga Barang      : Rp47500.0

=== HAK AKSES PELANGGAN ===
Kode Hak Akses              : 1111
Member Access               : True
Promo Access                : True
Free Shipping Access        : True

=== OPERASI BITWISE ===
=== Kode Status Transaksi ===
0001 | 0010 | 0100 | 1000
Kode Biner   : 1111
Kode Desimal : 15

=== Pemeriksaan Status ===
Cek Member
1111 & 0001
Hasil Biner   : 0001
Hasil Desimal : 1

Cek Promo
1111 & 1000
Hasil Biner   : 1000

<div style="font-family: 'Times New Roman', Times, serif;font-size: 18px;">

Berikut adalah penjelasan alur logika program dalam bentuk **teks bersih** (tanpa format Markdown/HTML) agar mudah di-copy-paste langsung ke dokumen laporan praktikum Anda:

Penjelasan Alur Logika Program (tugas3_1011.py)

1. Menampilkan Judul Program <br>
print("=== SISTEM TRANSAKSI TOKO ===")
Menampilkan header atau judul utama dari sistem transaksi toko pada layar terminal.
2. Memasukkan Data Pelanggan <br>
nama_1011 = input("Masukkan Nama Pelanggan : ")
status_1011 = input("Masukkan Status Pelanggan (member/nonmember) : ")
total_belanja_1011 = int(input("Masukkan Total Belanja : "))
jumlah_barang_1011 = int(input("Masukkan Jumlah Barang : "))
kode_promo_1011 = input("Masukkan Kode Promo : ")
Program mengambil masukan dari pengguna secara interaktif. Fungsi int() digunakan untuk mengonversi nilai input total belanja dan jumlah barang yang awalnya berupa teks (string) menjadi bilangan bulat (integer) agar dapat dihitung matematis.
3. Menentukan Kode Promo yang Tersedia <br>
daftar_promo_1011 = ["SABTUMAJU", "MINGGUSERU", "LAPANDELAPAN", "HEMATBERSAMA"]
Mendefinisikan sebuah list yang berisi daftar kode promo resmi yang berlaku dalam sistem.
4. Evaluasi Kondisi Transaksi <br>
is_member_1011 = (status_1011.lower() == "member")
is_cukup_belanja_1011 = total_belanja_1011 >= 200000
is_cukup_barang_1011 = jumlah_barang_1011 >= 3
is_promo_tersedia_1011 = kode_promo_1011 in daftar_promo_1011
Mengevaluasi kondisi transaksi menggunakan operator perbandingan (==, >=) dan operator keanggotaan (in). Metode .lower() digunakan untuk memastikan pengecekan status tidak terpengaruh oleh huruf besar atau kecil. Hasil dari setiap evaluasi variabel ini berupa nilai Boolean (True atau False).
5. Evaluasi Kombinasi Logika <br>
dapat_diskon_1011 = is_cukup_belanja_1011 or is_cukup_barang_1011
dapat_promo_1011 = is_promo_tersedia_1011
Menggunakan operator logika or untuk menentukan apakah pelanggan berhak menerima diskon, yaitu jika total belanja minimal Rp200.000 atau jumlah barang yang dibeli minimal 3 buah.
6. Menghitung Potongan dan Total Pembayaran <br>
persen_diskon_1011 = 0.05
potongan_harga_1011 = total_belanja_1011 * persen_diskon_1011
total_pembayaran_1011 = float(total_belanja_1011)
total_pembayaran_1011 -= potongan_harga_1011
rata_rata_harga_1011 = total_pembayaran_1011 / jumlah_barang_1011
Menghitung besarnya diskon sebesar 5% dari total belanja dengan operator perkalian (*). Total pembayaran bersih diperbarui nilainya menggunakan operator penugasan augmented assignment (-=). Rata-rata harga per barang dihitung dengan membagi total pembayaran setelah diskon dengan jumlah barang menggunakan operator pembagian (/).
7. Pembentukan Bitmask dan Gabungan Bitwise <br>
BIT_MEMBER_1011 = 1 if is_member_1011 else 0
BIT_BELANJA_1011 = 2 if is_cukup_belanja_1011 else 0
BIT_BARANG_1011 = 4 if is_cukup_barang_1011 else 0
BIT_PROMO_1011 = 8 if is_promo_tersedia_1011 else 0
kode_transaksi_1011 = BIT_MEMBER_1011 | BIT_BELANJA_1011 | BIT_BARANG_1011 | BIT_PROMO_1011
kode_referensi_1011 = BIT_MEMBER_1011 | BIT_BELANJA_1011 | BIT_PROMO_1011
Menentukan nilai desimal representasi bit untuk tiap kondisi (1 untuk bit-0, 2 untuk bit-1, 4 untuk bit-2, dan 8 untuk bit-3). Operator Bitwise OR (|) digunakan untuk menggabungkan seluruh flag status tersebut menjadi satu nilai angka unik berupa kode_transaksi_1011 dan kode_referensi_1011.
8. Evaluasi Bitwise Lanjutan <br>
cek_member_1011 = kode_transaksi_1011 & 0b0001
cek_promo_1011 = kode_transaksi_1011 & 0b1000
hasil_xor_1011 = kode_transaksi_1011 ^ kode_referensi_1011
hasil_shift_1011 = kode_transaksi_1011 << 1
Melakukan serangkaian operasi bitwise:

* Operator Bitwise AND (&) digunakan untuk bitmasking dalam menguji apakah bit status member atau promo bernilai aktif.
* Operator Bitwise XOR (^) digunakan untuk membandingkan perbedaan status bit antara kode transaksi dan kode referensi.
* Operator Bitwise Left Shift (<<) digunakan untuk menggeser posisi bit kode transaksi ke kiri sebanyak 1 langkah.

9. Menampilkan Data Transaksi dan Hasil Validasi <br>
print("\n=== DATA TRANSAKSI ===")
...
print("\n=== HASIL VALIDASI ===")
Mencetak kembali ringkasan rincian data transaksi milik pelanggan beserta seluruh hasil validasi syarat dalam bentuk Boolean (True/False).
10. Menampilkan Hasil Perhitungan Aritmatika <br>
print(f"Diskon                      : Rp{potongan_harga_1011:.1f}")
print(f"Total Pembayaran            : Rp{total_pembayaran_1011:.1f}")
print(f"Rata-rata Harga Barang      : Rp{rata_rata_harga_1011:.1f}")
Menampilkan rincian keuangan berupa nominal potongan harga, total pembayaran akhir, dan rata-rata harga barang dengan format pembulatan 1 angka di belakang koma (:.1f).
11. Menampilkan Informasi Hak Akses <br>
print(f"Kode Hak Akses              : {bin(kode_transaksi_1011)[2:].zfill(4)}")
print(f"Member Access               : {is_member_1011}")
print(f"Promo Access                : {is_promo_tersedia_1011}")
print(f"Free Shipping Access        : {kode_promo_1011 == 'SABTUMAJU'}")
Menampilkan status hak akses pelanggan, termasuk validasi akses gratis ongkir khusus untuk pengguna dengan kode promo "SABTUMAJU".
12. Menampilkan Detail Kode Status Bitwise <br>
print(f"{BIT_MEMBER_1011:04b} | {BIT_BELANJA_1011:04b} | {BIT_BARANG_1011:04b} | {BIT_PROMO_1011:04b}")
print(f"Kode Biner   : {kode_transaksi_1011:04b}")
print(f"Kode Desimal : {kode_transaksi_1011}")
Menggunakan f-string format specifier :04b untuk mengubah nilai desimal menjadi string biner berpanjang 4 bit secara otomatis dan rapi.
13. Menampilkan Hasil Operasi Bitwise Lengkap <br>
print(f"{kode_transaksi_1011:04b} & 0001")
print(f"Hasil Biner   : {cek_member_1011:04b}")
print(f"{kode_transaksi_1011:04b} ^ {kode_referensi_1011:04b}")
print(f"Hasil Biner   : {hasil_xor_1011:04b}")
print(f"{kode_transaksi_1011:04b} << 1")
print(f"Hasil Biner   : {hasil_shift_1011:05b}")
Mencetak visualisasi proses perhitungan bitwise AND, XOR, serta Left Shift beserta hasil keluaran bentuk biner dan desimalnya. Hasil pergeseran bit ditayangkan dalam format 5 bit (:05b) untuk menampung digit baru.
14. Menampilkan Tanda Selesai <br>
print("=== SELESAI ===")
Menandakan bahwa alur eksekusi seluruh program telah selesai dijalankan.

</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">BAB III<br>KESIMPULAN</h1>

Berdasarkan praktikum yang telah dilaksanakan, dapat disimpulkan bahwa operator merupakan bagian penting dalam pemrograman Python karena digunakan untuk melakukan berbagai macam proses terhadap data. Operator yang dipelajari meliputi operator aritmatika, penugasan, perbandingan, logika, keanggotaan, identitas, dan bitwise, yang masing-masing memiliki fungsi serta penggunaan yang berbeda.

Melalui praktikum ini, mahasiswa dapat memahami cara kerja dan penerapan setiap operator dalam program sederhana. Pemahaman tersebut dapat membantu mahasiswa dalam melakukan perhitungan, membandingkan nilai, mengolah kondisi, memeriksa keanggotaan dan identitas objek, serta melakukan operasi pada bit. Dengan demikian, penguasaan operator Python menjadi dasar yang penting untuk mempelajari pemrograman pada tahap selanjutnya.

</div>

---
<div style="height:1123px;font-family: 'Times New Roman', Times, serif;font-size: 18px;">
<h1 align="center">DAFTAR PUSTAKA</h1>

1. Python Software Foundation. (2026). Python 3 Documentation. Python Software Foundation. https://docs.python.org/3/

2. Python Software Foundation. (2026). The Python Tutorial. Python Software Foundation. https://docs.python.org/3/tutorial/

3. W3Schools. (2026). Python Operators. W3Schools. https://www.w3schools.com/python/python_operators.asp

4. GeeksforGeeks. (2026). Python Operators. GeeksforGeeks. https://www.geeksforgeeks.org/python/python-operators/
</div>